components/channel_view/channel_view.jsx

```diff
@@ -43,18 +43,31 @@ export default class ChannelView extends React.PureComponent {
     }
 
     static getDerivedStateFromProps(props, state) {
-        let updatedState = {url: props.match.url};
-        if (!state.url || props.match.url !== state.url) {
-            updatedState = {...updatedState, deferredPostView: ChannelView.createDeferredPostView()};
+        let updatedState = {};
+        if (props.match.url !== state.url) {
+            updatedState = {deferredPostView: ChannelView.createDeferredPostView(), url: props.match.url};
```

Comment:
Not important for this PR, but I just had a thought that we might be able to get rid of this state by using a key to on the `DeferredPostView` to force it to re-render whenever the channel changes